### PR TITLE
Adjust outbound shipment allocation and display info message

### LINKED ISSUE #1717: The packsize listbox is not updated with the right packsize of the stock allocated
## What went wrong? 😲
 
As title. 

packsize list not updated

## Expected behaviour 🤔
I would expect that if one line is chosen then the packsize of that line would be selected. If many lines of different packsize are used then it should be any. 

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Create an internal order with lines with quantity lower than all the packsize available for that item at the supplier store
2. In the customer requisition created, supply the stock and create an outbound shipment.
3. Click on a line to allocate stock. Put the quantity equal to a packsize you have instead of the requested quantity. Click on allocate  
4. See error. No packsize has been selected in the list.

## Your environment 🌱

 - Version: 
 - Platform:
   - [ ] android (tablet)
   - [ ] browser (extra points if you tell us which one)
   - [X] desktop (windows)
   - [ ] desktop (macOS)
   - [ ] server (windows)

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #1717 

# 👩🏻‍💻 What does this PR do? 
 
Allows over-allocation when the pack size requires it. When this happens, displays a message to the user warning them.

I've updated the test suite to accomodate the change in logic and have simplified the allocation routine in utils, didn't need to be as complex as it was.

One strange thing in the original issue which I could not replicate - the pack size controller should only show `100` if all stock lines have pack size of `100`. For my tests, I've added a stock line with pack size of `1` so that the `Any` option is available. This change only applies when `Any` is used, and units are being allocated.

A demo of the process:

https://github.com/openmsupply/open-msupply/assets/9192912/e9c41636-be0c-407c-b8b5-145f9d2605c0

# 🧪 How has/should this change been tested? 

`yarn test packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/utils.test.ts`

and some manual testing of the info panel in outbound line edits.

## 📃 Documentation

- [ ] Add the warning to the editing outbound modal bit

### CODE REVIEW

File: client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/utils.ts [hunk 1]
```diff
@@ -235,30 +199,26 @@ const reduceBatchAllocation = ({
...
 
       newDraftOutboundLines[draftOutboundLineIdx] = {
         ...draftOutboundLine,
         numberOfPacks: numberOfPacks,
-        isUpdated: true,
+        isUpdated: numberOfPacks > 0,
       };
     });
-  return Math.abs(toAllocate);
+  return -1 * toAllocate;
```
Comment (CarlosNZ) [inline client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/utils.ts:221]: Or just `-toAllocate`?

**Review by CarlosNZ — COMMENTED** [review]
Cool, LGTM .

I don't fully understand this aspect of the app, so haven't been overly thorough looking through, but I won't hold you up on the build.

Tests pass, can't see anything obviously incorrect,  so... 👍

**Review by CarlosNZ — APPROVED** [review]
Oops, selected "Comment" the first time. "Approved" now :)

File: client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/utils.ts (hunk 1 above)
Comment (mark-prins) [reply client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/utils.ts:221]: nice! thanks

**Pull request merged**